recipes-bsp/bootfiles/rpi-cmdline.bb

```diff
@@ -29,12 +29,65 @@ CMDLINE_LOGO ?= '${@oe.utils.conditional("DISABLE_RPI_BOOT_LOGO", "1", "logo.nol
 # to enable kernel debugging.
 CMDLINE_DEBUG ?= ""
 
-# Add RNDIS capabilities (must be after rootwait)
-# example: 
-# CMDLINE_RNDIS = "modules-load=dwc2,g_ether g_ether.host_addr=<some MAC 
-# address> g_ether.dev_addr=<some MAC address>"
+# Add gadget capabilities (must be after rootwait)
 # if the MAC addresses are omitted, random values will be used
-CMDLINE_RNDIS ?= ""
+def setup_gadget_mac(d, gadget_type):
+    string = ""
+    if d.getVar('GADGET_HOST_MAC_ADDR'):
+        string += " " + gadget_type + ".host_addr="\
+          + d.getVar('GADGET_HOST_MAC_ADDR')
+    if d.getVar('GADGET_DEV_MAC_ADDR'):
+        string += " " + gadget_type + ".dev_addr="\
+          + d.getVar('GADGET_DEV_MAC_ADDR')
+    return string
+
+def setup_mass_storage(d, gadget_type):
+    string = ""
+    if d.getVar('GADGET_MASS_STORAGE_NAME'):
+        string += " " + gadget_type + ".file="\
+          + d.getVar('GADGET_MASS_STORAGE_NAME')
+    return string
+
+# Setup the gadget configuration
+def setup_gadget_mode(d):
+    string = ""
+    if not bb.utils.contains('DISTRO_FEATURES', 'systemd', True, False, d)\
+        or d.getVar('ENABLE_DWC2_PERIPHERAL') != "1":
```

Comment:
Could we add an incompatibility warning/error? Otherwise, configuration might get ignored silently.